### PR TITLE
Move process in prj/processes/process

### PR DESCRIPTION
**Pull request by endJunction**
Aside moving files around and adding the `prj__processes__process` to everything related to processes there is a small change on when the process variables are parsed in the process creation.
See commit "[PL] Move process_variables_parsing in own process"

If you like to review everything, then commit-wise is better.

### CODE REVIEW

File: scripts/doc/append-xml-tags.py [hunk 1]
```diff
@@ -35,6 +35,8 @@
 
 # used to expand documentation entry points to full xml tag paths
 # that are used in the prj file.
+# For example process on top-level could be expanded to processes.process.
+# See the expansion table in the linked-xml-file.py too.
 tag_path_expansion_table = {
     "process":            "processes.process",
```
Comment (chleh) [inline scripts/doc/append-xml-tags.py]: Shouldn't this now be removed, too?

File: ProcessLib/LiquidFlow/CreateLiquidFlowProcess.cpp [hunk 2]
```diff
@@ -32,15 +32,19 @@ std::unique_ptr<Process> createLiquidFlowProcess(
     unsigned const integration_order,
     BaseLib::ConfigTree const& config)
 {
-    //! \ogs_file_param{process__type}
+    //! \ogs_file_param{prj__processes__process__type}
```
Comment (wenqing) [inline ProcessLib/LiquidFlow/CreateLiquidFlowProcess.cpp]: Should I update PR #1591 according to the changes in this PR?

Comment (endJunction) [inline ProcessLib/LiquidFlow/CreateLiquidFlowProcess.cpp]: I've update these just now. You might have a look on those changes.

**Comment by endJunction** [comment]
Jenkins, test this please!

**Comment by endJunction** [comment]
@bilke Somehow the build directory in jenkins is not good for merges. It complains about
```
Merging master commit b262a4bb6b5f54e674fe474b45294e0c8b7d14b8 into PR head commit 07d64e1be75382823fb39c71c420c4f99038e72d
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 07d64e1be75382823fb39c71c420c4f99038e72d
 > git merge b262a4bb6b5f54e674fe474b45294e0c8b7d14b8 # timeout=10
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 07d64e1be75382823fb39c71c420c4f99038e72d

GitHub has been notified of this commit’s build result

hudson.plugins.git.GitException: Command "git merge b262a4bb6b5f54e674fe474b45294e0c8b7d14b8" returned status code 128:
stdout: 
stderr: error: merge is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: Exiting because of an unresolved conflict.
```

Although there is nothig to merge. I tried to delete builds, restart, etc. with no success.

I'll keep this PR open for a while, so the logs and build directories are not deleted, but I'll open another PR, which hopefully works fine.

**Comment by endJunction** [comment]
GitHub was to clever to close this PR.

**Pull request merged**